plugins/sql_extract/src/sql_extract/component.py

```
@@ -0,0 +1,264 @@
+        if scheme in scheme_mapping:
+            # Заменяем схему на асинхронную версию
+            new_scheme = scheme_mapping[scheme]
+            return uri.replace(f"{scheme}://", f"{new_scheme}://", 1)
+
+        return uri
+
+    async def _execute_query(self, query: str) -> pl.DataFrame | None:
+        """Выполнение SQL запроса через SQLAlchemy."""
+        if not self._session_factory:
+            msg = "Session factory не инициализирована"
+            raise Exception(msg)
```

Ruff (0.12.2)

Create your own exception

(TRY002)